### DISCUSSION TITLE
how to set the parameter of strata in adonis2

### DISCUSSION THREAD

**Discussion by Ivy-ops** [vegandevs/vegan#600] [Q&A] [answered]
Hi developer, 
I have been searching the whole website regarding how to set the "strata" in adonis2(), but it seems there is no confirmed answer regarding this. Could you guide me on the below 2 designs?
(1) paired design: for example 30 samples from 15 subjects, each subject has pre and post-treatment(described by "treatment" variable). Is that accurate to set: adonis2(distance ~ treatment, strata = subject)? 
(2) repeated design: for example 30 samples from 10 subjects, each subject has 3 samples, and 5 subjects(5x3 = 15 samples) are treatment, 5 subjects(5x3 = 15 samples) are control. I want to compare treatment vs. control. Under this condition, how to set the formula? (assume "group" variable describes treatment vs. control, "subject" describes subject)? 
Much appreciated!

**Comment by gavinsimpson** [comment]
For 1, you can do that {with `strata`], but I think it is best to learn how to define restricted permutations using the *permute* package. For a design-based fit you would create:

```
library("permute")
h1 <- with(your_data, how(blocks = subject))
```

and then pass `h1` to the `permutations` argument of `adonis2()` (if you must use that)

For 2, are your samples equally-spaced in time?

**Reply by Ivy-ops** [reply]
@gavinsimpson Thank you! 
For 1, do you mean: 

```
h1 <- with(data, how(blocks = subject)); 
adonis2(distance ~ treatment, data = data, permutations = h1, strata =data$subject)
```

Is this correct? Here treatment is time (pre-treatment and post-treatment).

For 2, it is a repeated design, with no time variable. 30 samples from 10 subjects(each subject has 3 samples), 5 subjects belong to control (which means 5*3 =15 samples belong to control), and the rest 15 samples from the other 5 subjects belong to treatment. How to do that in adonis2()?

Looking forward to your thoughts!

**Reply by gavinsimpson** [reply]
No, you don't use `strata` at all - you specified the blocking structure needed in the call to `how()`

**Reply by Ivy-ops** [reply]
Thank you @gavinsimpson . 
So for 1: 
h1 <- with(data, how(blocks = subject)); 
adonis2(distance ~ treatment, data = data, permutations = h1, strata =NULL)
For 2: Do you have any thoughts on the repeated design? Much appreciated!

**Reply by gavinsimpson** [reply, marked as answer]
You don't need to specify anything for `strata` &mdash; just don't even mention it:

```r
h1 <- with(your_data, how(nperm = 999, blocks = subject))
adonis2(distance ~ treatment, data = your_data, permutations = h1)
```

However I would suggest you do this via `rda()` because there is no easy way to *exclude* the between `subject` variation:

```r
h1 <- with(your_data, how(nperm = 999, blocks = subject))
ord <- rda(response_df ~ treatment + Condition(subject), data = your_data)
anova(ord, permutations = h1)
```

would stop any between `subject` variation getting counted in the `treatment` effect.

As for your second design, you need to look at where the thing you want to test varies. Clearly it doesn't vary at the within-`subject` level: each `subject` only ever receives one of the treatment levels. So those samples shouldn't ever be permuted. Where the `treatment` varies is at the between-`subject` level. As such, under the null hypothesis of no effect of `treatment` and conditional upon us not permuting the samples *within* a `subject`, the observations at the *between*-`subject` are exchangeable. This means you want to permute the `subject`s, which is achieved through the use of the `plots` argument to `how()`:

```r
h2 <- with(your_data, how(nperm = 999,
  plots = Plots(strata = subject, type = "free"),
  within = Within(type = "none")))
```

then you fit your model as

```r
adonis2(distance ~ treatment, your_data = data, permutations = h2)

# or
ord <- rda(response_df ~ treatment + Condition(subject), data = your_data)
anova(ord, permutations = h2)
```

where again, conditioning on `subject` will remove the between-`subject` effect from the estimate of the `treatment` effect. In both cases I don't think it (the conditioning) is needed for the purposes of the permutation test, but if you ignore it, you might inflate the inertia ("variance", sums of squares, etc) attributable to the `treatment` effect when you report the inertia explained value.

**Reply by jarioksa** [reply]
Please not that `adonis2` defaults to Bray-Curtis distance but `rda` is based on Euclidean distance. For equivalent models you have two choices:
1. `adonis2(..., method="euclidean")` equals `anova(rda(...))` for Euclidean distance in both.
2. `adonis2(...)` equals `anova(dbrda(..., method="bray"))` for Bray-Curtis in both.

In general `adonis2()` and `anova(dbrda())` give identical results, although `adonis2` may be marginally faster. With same permutations (or `how` object) they should give identical permutation statistics. `adonis2` folds together fitting and testing, and with ordination (`dbrda`) permutation arguments should be given to `anova` so that these should be equivalent:
```r
adonis2(formula, data, permutations, method, sqrt.dist, add, by, parallel)
## or alternatively with dbrda
ORD <- dbrda(formula, data, method, sqrt.dist, add) 
anova(ORD, permutations, by, parallel)
```

**Reply by Ivy-ops** [reply]
Thank you both @gavinsimpson @jarioksa! Your answers have been the most detailed I've received!

**Comment by jarioksa** [comment]
Please note that `strata` is a legacy argument in **vegan**. It was copied from the **boot** package over 20 years ago, and provides only the simplest blocking of permutations. You really need to use **permute** package for anything more sophisticated. I was about to remove the `strata` argument completely in favour of **permute** package, but I was convinced that it is better to maintain backward-compatibility and  `strata` is sufficient for many a need. Actually, it was first dropped from `adonis2`, but people still *used* it although it was not there and was non-functional – the `strata` were just silently ignored. I decided it is best to put it back so that it works if people use it. The reason that you didn't find information about implementing your design with `strata` was that you cannot: you must see how to do these things with **permute** that is internally used in **vegan**.

**Reply by Ivy-ops** [reply]
Thank you! @jarioksa

**Comment by D-rgz** [comment]
Hi developer I want to do permutational analysis of variance to calculate variation between nematode haplotypes of the same species in samples , i have unbalanced haplotypes in samples and in regions (108 samples, 3 regions, sharing 27 haplotypes). Each sample has a range of between 1 to 9  haplotypes, I have built a pairwise distance matrix (hamming) for all haplotypes. I want to build an R script for performing analysis of molecular variance within samples and between samples in a regions and between regions (I have 3 regions). I guess my challenge is to build the correct dataframe shape ..i tried the nested design making Region and SampleID  for each Haplotype so the rows match the dist matrix but it wouldnt run, and also i realize that i have to build it using the **permute**  function over samples??? I also want to do an ordination for the different sample to show how they vary between the 3 regions
Your help will be much appreciated.

**Reply by gavinsimpson** [reply]
Please don't jump onto an existing discussion; start a new discussion question instead. If you do, please note that you will need to explain what you want to do and your setting with more detail otherwise we can't possibly help.

**Comment by D-rgz** [comment]
Thanks I will
Deo
________________________________

**Comment by Ivy-ops** [comment]
Hi @gavinsimpson and @jarioksa,

It’s me again — I have a few follow-up questions related to my previous post. I hope it’s okay to continue the discussion here; if not, I can start a new thread.

Regarding my earlier question about a paired design (e.g., 30 samples from 15 subjects, each with pre- and post-treatment measurements indicated by the variable treatment), thanks for the suggestions on testing the treatment effect, which is pretty helpful!

However, if I now want to examine whether **_the change between pre- and post-treatment_** is associated _**with another continuous variable (e.g., BMI)**_, can this be tested directly using adonis2() or rda()? I am thinking below?
```
h1 <- with(your_data, how(nperm = 999, blocks = subject))
adonis2(distance ~ treatment+BMI+treatment:BMI, data = your_data, permutations = h1)
```
I would greatly appreciate your thoughts on the appropriate approach.

Best regards,
-Ivy
